client/verta/tests/unit_tests/pipeline/test_pipeline_step.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for the PipelineStep class
+"""
+
+import random
+
+from hypothesis import given, HealthCheck, settings
+
+from tests.unit_tests.strategies import mock_pipeline_definition
+from verta.pipeline import PipelineStep
+
+
+@given(pipeline_definition=mock_pipeline_definition())
+@settings(
+    suppress_health_check=[HealthCheck.function_scoped_fixture],
+    deadline=None,
+)
+def test_steps_from_pipeline_definition(
+    pipeline_definition,
+    mock_conn,
+    mock_config,
+    mocked_responses,
+) -> None:
+    """
+    Test that a list of PipelineStep objects can be constructed and returned from
+    a pipeline definition.  The registered model, model version, and environment
+    is fetched for each step, so a response is mocked for each call.
+    """
+    graph = pipeline_definition["graph"]
+    for step in pipeline_definition["steps"]:
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/model_versions/{step['model_version_id']}",
+            json={"model_version": step["model_version_id"]},
+            status=200,
+        )
+        mocked_responses.get(
+            f"https://test_socket/api/v1/registry/registered_models/0",
+            json={},
+            status=200,
+        )
+    generated_steps = PipelineStep._steps_from_pipeline_definition(
+        pipeline_definition=pipeline_definition,
+        conn=mock_conn,
+        conf=mock_config,
+    )
+    # we have the same number of steps as in the pipeline definition
+    assert len(generated_steps) == len(pipeline_definition["steps"])
+    for spec_step, gen_step in zip(pipeline_definition["steps"], generated_steps):
+        # each step is converted to a PipelineStep object
+        assert isinstance(gen_step, PipelineStep)
+        # the names are the same for the steps and their definitions
+        assert gen_step.name == spec_step["name"]
+        # predecessors for each step are also converted to PipelineStep objects
+        for i in gen_step.predecessors:
+            assert isinstance(i, PipelineStep)
+        # the predecessors for each step are the same as in the definition
+        assert set([i.name for i in gen_step.predecessors]) == set(
+            [s["predecessors"] for s in graph if gen_step.name == s["name"]][0]
+        )
+
+
+def test_to_step_spec(make_mock_registered_model_version) -> None:
+    """Test that a PipelineStep object can be converted to a step specification"""
+    mod_version = make_mock_registered_model_version()
```

Comment:
Can we use either `model_version`, `model_ver`, or `mv`? We don't use the term `mod_version` anywhere else in the repo

Reply:
Done